mmv1/third_party/terraform/tests/resource_dialogflowcx_environment_test.go

```diff
@@ -0,0 +1,121 @@
+			},
+			{
+				ResourceName:      "google_dialogflow_cx_environment.development",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccDialogflowCXEnvironment_full(context),
+			},
+			{
+				ResourceName:      "google_dialogflow_cx_environment.development",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccDialogflowCXEnvironment_basic(context map[string]interface{}) string {
+	return Nprintf(`
+	resource "google_service_account" "dialogflowcx_service_account" {
+		account_id = "tf-test-dialogflow-%{random_suffix}"
+	}
+	  
+	resource "google_project_iam_member" "agent_create" {
+		role    = "roles/dialogflow.admin"
+		member  = "serviceAccount:${google_service_account.dialogflowcx_service_account.email}"
+	}
+
+	resource "google_dialogflow_cx_agent" "agent_version" {
+		display_name = "tf-test-%{random_suffix}"
+		location = "global"
+		default_language_code = "en"
+		supported_language_codes = ["fr","de","es"]
+		time_zone = "America/New_York"
+		description = "Description 1."
+		avatar_uri = "https://storage.cloud.google.com/dialogflow-test-host-image/cloud-logo.png"
+		depends_on = [google_project_iam_member.agent_create]
+	}
+    
+	resource "google_dialogflow_cx_version" "version1" {
+		parent       = google_dialogflow_cx_agent.agent_version.start_flow
+		display_name = "1.0.0"
+		description  = "version 1.0.0"
+	}
+
+	resource "google_dialogflow_cx_environment" "development" {
+        parent       = google_dialogflow_cx_agent.agent_version.id
+        display_name = "Development"
+        version_configs {
+            version = google_dialogflow_cx_version.version1.id
+        }
+    }
+    `, context)
+}
+
+func testAccDialogflowCXEnvironment_full(context map[string]interface{}) string {
+	return Nprintf(`
+    resource "google_service_account" "dialogflowcx_service_account" {
+		account_id = "tf-test-dialogflow-%{random_suffix}"
+	}
+	  
+	resource "google_project_iam_member" "agent_create" {
+		role    = "roles/dialogflow.admin"
+		member  = "serviceAccount:${google_service_account.dialogflowcx_service_account.email}"
+	}
+
+	resource "google_dialogflow_cx_agent" "agent_version" {
+		display_name = "tf-test-%{random_suffix}update"
+		location = "global"
+		default_language_code = "en"
+		supported_language_codes = ["no"]
+		time_zone = "Europe/London"
+		description = "Description 2!"
+		avatar_uri = "https://storage.cloud.google.com/dialogflow-test-host-image/cloud-logo-2.png"
+		enable_stackdriver_logging = true
+        enable_spell_correction    = true
+		speech_to_text_settings {
+			enable_speech_adaptation = true
+		}
+		depends_on = [google_project_iam_member.agent_create]
+	}
+
+	resource "google_dialogflow_cx_version" "version1" {
+		parent       = google_dialogflow_cx_agent.agent_version.start_flow
+		display_name = "1.0.0"
+		description  = "version 1.0.0"
+	}
+
+	resource "google_dialogflow_cx_environment" "development" {
```

Comment:
Similar to my comment on the other PR, can you update the environment here while leaving the `google_dialogflow_cx_agent` the same? This will test that the update method on the environment works as expected. 

Ideally we would add a version, then remove a version as that seems to be the important part of this resource

Reply:
Got it, I updated tests so basically test will publish version 1 and then environment will be updated with a new version 2